meson.build

```diff
@@ -252,10 +254,13 @@ if get_option('coverage') == true
     ]
 endif
 
+hwloc_dep = dependency('hwloc', static : dep_static)
+
 target_deps += [
     boost_dep,
     dl_dep,
     pthread_dep,
+    hwloc_dep,
```

Comment:
This portion of the change should probably be moved a few commits up, so that `<hwloc.h>` is found and linked to.

Nitpick: `hwloc_dep` sorts before `pthread_dep`